### ISSUE TITLE
NavGuards with async HTTP Calls does not work v3 and probably also v2

### ISSUE THREAD

**Issue by mburger81** [driftyco/ionic#11090]
**Ionic version:**  (check one with "x")
[ ] **1.x**
[x] **2.x**
[3] **2.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[x] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**
We are trying to handle a NavGuard calling a HTTP request by an Obersvable. We have an get for current/user in an service, something like that:
```
    current() {

        return this.http.get('/api/v1/users/current')
                        .map(res => res.json());
    }
```

for our lazy loading page on ionic3 we will check if user is authenticated on ionViewCanEnter, so this has to be async.

We are doing this:
```
ionViewCanEnter(): Promise<any> {
        return new Promise((resolve, reject) => {

                /*
                this.userService.current().toPromise()
                    .then(data => {console.log('data'); resolve()})
                    .catch(err => {console.log('err'); resolve()})
                */

                this.userService.current()
                        .subscribe(
                            (data) => {
                                console.log('User is: ', data);
                                resolve('OK');
                            },
                            (err) => {
                                console.log('User is NOT AUTHENTICATED: ', err);

                                //Redirect the user before denying them access to this route
                                //this.router.navigate(['/login']);
                                reject('ERR');
                            },
                            () => {
                                console.log('????');
                                reject('???');
                            });

            });
    }
```

The http call is never started, so it looks like there is now subscription to observable. It's difficult to debug, but I think it could be a problem by combine `Observables` with promises.
Also the commented code with `toPromise()` for observable does not work.

**Expected behavior:**
The thing is, it would be nice if we can call in NavGuards directly some httpCalls, but this for now we can not do. I don't know if this is a bug of Promise/Observable or Ionic.

**Other information:**

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

```
    Cordova CLI: 6.5.0 
    Ionic Framework Version: 3.0.0
    Ionic CLI Version: 3.0.0-beta.5
    ios-deploy version: Not installed
    ios-sim version: Not installed
    OS: Linux 4.8
    Node Version: v6.10.2
    Xcode version: Not installed
```

**Comment by jgw96** [comment]
Hello, thanks for using Ionic! We will look into this.

**Comment by zbarbuto** [comment]
@mburger81 What happens if you change it to the following:

```typescript
ionViewCanEnter(): Promise<any> {
  return this.userService.current().toPromise();
}
```

**Comment by mburger81** [comment]
@zbarbuto It works but there is also a BUG on it
I'm not sure when this is happening but on doing this I get an error on some situation
This is an example code which alls an Obersvable and get a promise from it, but in some situations I get an error see my attached ISSUE which I opened below
```
    ionViewCanEnter(): Promise<any> {
        return new Promise((resolve, reject) => {
                this.userService.current(true).toPromise()
                        .then((data) => {
                            resolve();
                        })
                        .catch((error) => {
                            reject('ERROR');
                        });
        });
    }
```

https://github.com/driftyco/ionic/issues/11168

**Comment by jgw96** [comment]
Hello! Thanks for opening an issue with us! As this seems like more of a support question related to using observables, I will urge that you ask this question on our forum (forum.ionicframework.com) or on our [slack channel](https://ionic-worldwide.slack.com/messages?]. Thanks for using Ionic!

**Comment by mburger81** [comment]
Comm on guys! This is not a support request. This was definitly a bug you  closed some release ago with version 3.0.1. We don't know if it was a side effect or something else.
